src/events/antiSpam/messageCreate.ts

```diff
@@ -0,0 +1,77 @@
+import { Client, Constants, Message } from 'discord.js';
+import { inject, injectable } from 'tsyringe';
+import type { Event } from '../../Event';
+import { kRedis } from '../../tokens';
+import { on } from 'node:events';
+import { createHash } from 'node:crypto';
+import type { Redis } from 'ioredis';
+import { transformHashset } from '../../util/redisData';
+import { logger } from '../../logger';
+import { CaseAction, createCase } from '../../functions/cases/createCase';
+import { upsertCaseLog } from '../../functions/logs/upsertCaseLog';
+import { getGuildSetting, SettingsKeys } from '../../functions/settings/getGuildSetting';
+import i18next from 'i18next';
+import { SPAM_EXPIRE_SECONDS, SPAM_THRESHOLD } from '../../Constants';
+
+@injectable()
+export default class implements Event {
+	public name = 'Spam check';
+
+	public event = Constants.Events.MESSAGE_CREATE;
+
+	public constructor(public readonly client: Client<true>, @inject(kRedis) public readonly redis: Redis) {}
+
+	public async execute(): Promise<void> {
+		for await (const [message] of on(this.client, this.event) as AsyncIterableIterator<[Message]>) {
+			try {
+				if (!message.guild || !message.content.length) continue;
+
+				const contentHash = createHash('md5').update(message.content).digest('hex');
+				const channelSpamKey = `guild:${message.guild.id}:user:${message.author.id}:contenthash:${contentHash}:channels`;
```

Comment:
Is there ever a scenario in which spam levels per channel would be useful? It seems like we could consolidate this into a single key to avoid summing all the values every time.

Reply:
I provide the list like this in my test implementation, requiring messages per channel (this bot acts on buttons rather than full auto moderation, making this information somewhat useful)

![image](https://user-images.githubusercontent.com/26532370/131261891-51e90a60-be7b-4393-982a-39c6818cdf7e.png)

I think you might be right, however, that for yuu's use case this makes less sense and could be more performant with a single val key

Reply:
going with a single value key